contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,204 @@
+# Encryption
```

Comment:
Generally I think the direction should be to make it as easy as possible to use a pluggable external secret store, and kubernetes should avoid re-implementing features, and accruing maintenance cost, of those stores. 

Adding encryption and associated key rotation as described here is a step down this re-implementation path. Probably followed by very detailed audit logging, followed by fancier authn/authz features. We need to draw a line under how far we want to travel down this path. If we want to include this, we should be explicit that it's as far as we're going re-inventing the wheel.

There's some danger that adding encryption here will tip some people into using k8s secrets when really the risk profile of what they are storing means they should be moving to an external store that has stronger audit trail guarantees and various other security features. Larger organizations will probably be motivated to use external stores regardless, because they are already storing other secrets there and just want to centralize on one system.

Re plugging in other stores, I think the right way to do this is to give pods (and perhaps nodes) identities they can use to talk to the stores directly, but this is out of scope for this proposal.

Re HA: Even if you are air-gapped and running your own vault installation you can make it HA with some work. The kubernetes piece of availability might be providing some flexibility in allowing nodes to access and cache HA secrets vs. the normal least-privilege path where only the pod can retrieve the secret. But this is also out of scope here.

I'm definitely still interested in meeting to talk about this.